besu/src/main/java/org/hyperledger/besu/controller/GasLimitCalculator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.hyperledger.besu.controller;
+
+import java.util.function.Function;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class GasLimitCalculator implements Function<Long, Long> {
+  private static final Logger LOG = LogManager.getLogger();
+  public static final long ADJUSTMENT_FACTOR = 1024L;
+  public static final Long DEFAULT = null;
+  private final long targetGasLimit;
+
+  public GasLimitCalculator(final Long targetGasLimit) {
```

Comment:
This should be an `OptionalLong`, we sholdn't be using null as a sentinel value.  our coding conventions doc calls for using modern java features such as Optional.

Reply:
You must be looking at outdated source code?

Reply:
Looks like github was giving me old code, they've been a bit wonky lately.